app/ide-desktop/lib/client/src/index.ts

```diff
@@ -34,6 +36,29 @@ const logger = contentConfig.logger
 /** Indent size for outputting JSON. */
 const INDENT_SIZE = 4
 
+/** Check if the given arguments show that we've been invoked with a file to open.
+ *
+ * For example, this happens when the user double-clicks on a file in the file explorer.
+ *
+ * @returns The path to the file to open, or `null` if no file was specified.
+ **/
+export function attemptingToOpenFile(clientArgs: string[]): string | null {
+    // If we are invoked with exactly one argument and this argument is a file, we assume that we have been
+    // invoked with a file to open. In this case, we must translate this path to the actual argument that'd open the
+    // project containing this file.
+    if (clientArgs.length === 1 && clientArgs[0] !== undefined) {
+        try {
+            fsSync.accessSync(clientArgs[0])
```

Comment:
if you extract `const arg = cleintArgs[0]`, then several next lines will be simplified.

Reply:
Fixed.